test/rest_api/test_storage_service.py

```diff
@@ -555,3 +555,34 @@ def test_storage_service_get_natural_endpoints(cql, rest_api, tablets_enabled):
             resp.raise_for_status()
 
             assert resp == [rest_api.host]
+
+@pytest.mark.xfail(reason="rest_api suite doesn't support tablets yet (#17338), run test manually")
+def test_range_to_endpoint_map_tablets_enabled_keyspace_param_only(cql,  rest_api):
+    with new_test_keyspace(cql, "WITH REPLICATION = { 'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1 } AND TABLETS = { 'enabled': true }") as keyspace:
+        with new_test_table(cql, keyspace, 'p int PRIMARY KEY') as table:
+            resp = rest_api.send("GET", f"storage_service/range_to_endpoint_map/{keyspace}")
+            assert resp.status_code == requests.codes.bad_request
+
+            resp_json = resp.json()
+            actual_error_reason = resp_json["message"]
+            expected_error_reason = f"storage_service/range_to_endpoint_map is per-table in keyspace '{keyspace}'. Please provide table name using 'cf' parameter."
+            assert expected_error_reason == actual_error_reason
+
+@pytest.mark.xfail(reason="rest_api suite doesn't support tablets yet (#17338), run test manually")
+@pytest.mark.parametrize("tablets_enabled", ["true", "false"])
+def test_range_to_endpoint_map_with_table_param(cql,  rest_api, tablets_enabled):
+    with new_test_keyspace(cql, f"WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1 }} AND TABLETS = {{ 'enabled': {tablets_enabled} }}") as keyspace:
+        with new_test_table(cql, keyspace, 'p int PRIMARY KEY') as table:
+            cf = table.split('.')[1]
+            resp = rest_api.send("GET", f"storage_service/range_to_endpoint_map/{keyspace}", params={"cf": cf})
```

Comment:
This api parameter is added only in the next patch, so let's add the test case along with it to maintain bisectability.

Reply:
Okay - the commits have been squashed. Now it is a single patch.

Even the commit message contained duplication, so it looks better now :+1: